electrumsv/daemon.py

```diff
@@ -312,7 +353,27 @@ def run_cmdline(self, config_options: dict) -> Any:
         result = func(*args, **kwargs)
         return result
 
+    def on_stop(self):
+        if self.rest_server and self.rest_server.is_alive:
+            app_state.async_.spawn_and_wait(self.rest_server.stop)
+        self.logger.debug("stopped.")
+
+    def launch_restapi(self):
+        if not self.rest_server.is_alive:
+            self._restapi_future = app_state.async_.spawn(self.rest_server.launcher)
+            self.rest_server.is_alive = True
+
     def run(self) -> None:
+        if app_state.config.get("cmd") == "daemon":
```

Comment:
This to me looks like it is reproducing something that could be done via app API.
e.g.
```
def run(self) -> None:
    app_state.app.some_started_up_event.wait() # Even this could be in ..app.wait_until_started()
    self.launch_restapi()
```

Reply:
Sure. Good idea.

Reply:
I have pushed the change for this - not sure if it is what you had in mind or not. 
I needed to add an "is_running" attribute to DefaultApp.

I would expect daemon apps to inherit from DefaultApp such that `app_state.app.wait_until_started()` method will **exist on the dapp**. They would then override and modify the run_app() as a template (but should set is_running = True before entering the main loop).